tensorflow_datasets/structured/movielens.py

```diff
@@ -0,0 +1,422 @@
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""MovieLens dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import csv
+import os
+import textwrap
+import codecs
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@article{10.1145/2827872,
+author = {Harper, F. Maxwell and Konstan, Joseph A.},
+title = {The MovieLens Datasets: History and Context},
+year = {2015},
+issue_date = {January 2016},
+publisher = {Association for Computing Machinery},
+address = {New York, NY, USA},
+volume = {5},
+number = {4},
+issn = {2160-6455},
+url = {https://doi.org/10.1145/2827872},
+doi = {10.1145/2827872},
+journal = {ACM Trans. Interact. Intell. Syst.},
+month = dec,
+articleno = {19},
+numpages = {19},
+keywords = {Datasets, recommendations, ratings, MovieLens}
+}
+"""
+
+_DESCRIPTION = """
+This dataset describes 5-star rating from MovieLens,
```

Comment:
> A couple of comments on the description, mostly about phrasing.
> 
> * The first line suggests this is a subset of all data that contains only 5-start ratings. If you mean that the datasets consists of movie ratings, something like the following might be more natural: "This dataset contains a set of movie ratings from MovieLens, a movie recommendation service."

Good point. I copied this sentence from the MovieLens homepage, but it is a bit misleading. I have changed the phrasing.

> * Perhaps you could mention that the ratings have been collected and the datasets are maintained by the GroupLens group at the University of Minnesota, and include a link?

I have added a line mentioning GroupLens. Regarding the link, did you mean a link to the MovieLens dataset homepage or the GroupLens homepage? A link to the MovieLens dataset homepage is already included in the returned DatasetInfo from _info. On most dataset documentation pages, the homepage link is listed separately from the description, so I think adding another link inside the description might be a bit redundant. I have added a link to the GroupLens homepage.

> * There seem to be many unnecessary line breaks here. Please use line breaks only when running against width limits or starting a new paragraph.

Got it. Fixed.

> * When describing the dataset variants, you could use a list to make it obvious. For example, you could write:
> 
> ```
> This dataset has the following variants:
> 
> - "25m": latest stable version...
> - "latest-small": subset of latest, ....
> ```
> 
> I suspect that the description gets interpreted as markdown, so this formatting should work well in the rendered docs as well.
> 

Great point. I have restructured this part of the description.

> * When you say "Users were selected at random for inclusion." it again suggests that this is a subset of the original Movielens dataset.

This sentence was also from the original MovieLens README. I have removed it since it does not convey useful information to the users.

> * Instead of mentioning dataset schemas, you could simply describe what information the datasets contain: as far as the user is concerned, the schema does not matter, and we present the information in the same way (it's just that some datasets have more information than others).

Could you clarify on what you mean by schema? Are you referring to parts like "the movies data and ratings data are joined on 'movieId'"? I have added a list of feature descriptions.